lib/gssapi/mech/gss_add_cred_from.c

```diff
@@ -162,45 +219,51 @@ gss_add_cred_from(OM_uint32 *minor_status,
         release_cred = (gss_cred_id_t)mut_cred;
     }
 
-    /* Find an MN, if any */
-    if (desired_name) {
-        major_status = _gss_find_mn(minor_status,
-                                    (struct _gss_name *)desired_name,
-                                    desired_mech, &mn);
-        if (major_status != GSS_S_COMPLETE)
-            goto done;
-    }
+    if (m->gm_flags & GM_USE_MG_CRED) {
+	struct _gss_mech_switch *ms;
+	OM_uint32 initiator_time_min = GSS_C_INDEFINITE;
+	OM_uint32 acceptor_time_min = GSS_C_INDEFINITE;
 
-    /*
-     * We go through all the mc attached to the input_cred_handle and check the
-     * mechanism.  If it matches, we call gss_add_cred for that mechanism,
-     * otherwise we just add a new mc.
-     */
-    HEIM_TAILQ_FOREACH(mc, &mut_cred->gc_mc, gmc_link) {
-        if (!gss_oid_equal(mc->gmc_mech_oid, desired_mech))
-            continue;
-        major_status = _gss_mg_add_mech_cred(minor_status, m,
-					     mc, mn, cred_usage,
-					     initiator_time_req, acceptor_time_req,
-					     cred_store, NULL,
-					     initiator_time_rec, acceptor_time_rec);
-        if (major_status != GSS_S_COMPLETE)
-            _gss_mg_error(m, *minor_status);
-        goto done;
-    }
+	major_status = GSS_S_UNAVAILABLE; /* in case of no mechs */
+
+	/*
+	 * Treat adding SPNEGO credentials (GM_USE_MG_CRED) as adding
```

Comment:
This is correct if `input_cred_handle == GSS_C_NO_CREDENTIAL`, otherwise, `if  input_cred_handle != GSS_C_NO_CREDENTIAL && (m->gm_flags & GM_USE_MG_CRED))` then this call is a no-op and we can just return `GSS_S_COMPLETE`.

Reply:
Thanks, sorry I missed this in your original patch. Will fix now.